filebeat/input/tcp/input.go

```diff
@@ -73,22 +74,28 @@ func NewInput(
 
 // Run start a TCP input
 func (p *Input) Run() {
-	if !p.started.Load() {
+	p.Lock()
+	defer p.Unlock()
+
+	if !p.started {
 		p.log.Info("Starting TCP input")
 		err := p.server.Start()
 		if err != nil {
 			p.log.Errorw("Error starting the TCP server", "error", err)
 		}
-		p.started.Swap(true)
+		p.started = true
 	}
 }
 
 // Stop stops TCP server
 func (p *Input) Stop() {
 	p.log.Info("Stopping TCP input")
```

Comment:
The logging is not congruent with the UDP input. In UDP the message is logged after acquiring the lock. IDK which order, but I would like the behavior to be the same.

Reply:
Correct, done in the next commit.